src/app/search/loadouts/search-filters/freeform.ts

```diff
@@ -22,6 +44,49 @@ const freeformFilters: FilterDefinition<
       return (loadout) => test(loadout.name);
     },
   },
+  {
+    keywords: ['subclass'],
+    description: tl('LoadoutFilter.Subclass'),
+    format: 'freeform',
+    suggestionsGenerator: ({ loadouts, d2Definitions }) => {
+      if (!loadouts || !d2Definitions) {
+        return [];
+      }
+      const damageDefs = damageDefsByDamageType(d2Definitions);
+      // TODO (ryan) filter on currently selected character. This info is currently localized
+      // to the page, so we need to lift that up before it can be done.
+      return deduplicate(
+        loadouts.flatMap((loadout) => {
+          const subclass = subclassDefFromLoadout(loadout, d2Definitions);
+          const damageType = subclass ? getDamageTypeForSubclassDef(subclass) : undefined;
+          // DamageType.None is 0
+          const damageName =
+            damageType !== undefined ? damageDefs[damageType]?.displayProperties.name : undefined;
+          return [
+            subclass
+              ? `subclass:${quoteFilterString(subclass.displayProperties.name.toLowerCase())}`
+              : undefined,
+            damageName ? `subclass:${quoteFilterString(damageName.toLowerCase())}` : undefined,
+          ];
+        }),
+      );
+    },
+    filter: ({ filterValue, language, d2Definitions }) => {
+      const test = matchText(filterValue, language, false);
+      const damageDefs = d2Definitions && damageDefsByDamageType(d2Definitions);
+      return (loadout: Loadout) => {
+        const subclass = d2Definitions ? subclassDefFromLoadout(loadout, d2Definitions) : undefined;
+        // DamageType.None is 0
+        const damageType = subclass ? getDamageTypeForSubclassDef(subclass) : undefined;
+        const damageName =
+          damageType !== undefined ? damageDefs?.[damageType]?.displayProperties.name : undefined;
+        return (
+          (subclass !== undefined && test(subclass.displayProperties.name)) ||
+          (damageName !== undefined && test(damageName))
+        );
```

Comment:
```suggestion
        const subclass = d2Definitions && subclassDefFromLoadout(loadout, d2Definitions);
        if (!subclass) { return false; }
        if (test(subclass.displayProperties.name)) { return true; }
        // DamageType.None is 0
        const damageType = getDamageTypeForSubclassDef(subclass)!;
        const damageName = damageDefs?.[damageType]?.displayProperties.name;
        return damageName && test(damageName));
```

Reply:
It will fail type check as the final return can return a string or undefined but I will fix it up.

This is happy the reason I used ternaries everywhere.